pkg/archive/zip.go

```diff
@@ -79,3 +81,98 @@ func ExtractZip(src string, dest string) error {
 	}
 	return nil
 }
+
+func ZipToTar(srcZip string) (string, error) {
+	tarFile, err := ioutil.TempFile("", "")
+	if err != nil {
+		return "", fmt.Errorf("create file for tar: %s", err)
+	}
+	defer tarFile.Close()
+
+	tw := tar.NewWriter(tarFile)
+	defer tw.Close()
+
+	zipReader, err := zip.OpenReader(srcZip)
+	if err != nil {
+		return "", err
+	}
+	defer zipReader.Close()
+
+	var fileMode int64
+	for _, f := range zipReader.File {
+		fileMode = -1
+		if isFatFile(f.FileHeader) {
+			fileMode = 0777
+		}
+
+		var header *tar.Header
+		if f.Mode()&os.ModeSymlink != 0 {
```

Comment:
Could this simplified with a golang switch?

Reply:
I think since it's only one if/else it wouldn't improve it too much